cds/modules/deposit/api.py

```diff
@@ -304,25 +304,20 @@ def _process_files(self, record_id, data):
 
             # create a copy of the deposit bucket for the record
             snapshot = self.files.bucket.snapshot()
+            snapshot.locked = False
             self._fix_tags_refs_to_master(bucket=snapshot)
             # dump after fixing references
-            self.files.bucket.locked = False
-            snapshot.sync(bucket=self.files.bucket, delete_extras=True)
-            self.files.bucket.locked = True
-            data["_files"] = self.files.dumps()
-
-            snapshot.locked = False
+            self.files.bucket.sync(bucket=snapshot, delete_extras=True)
```

Comment:
If I understand correctly the change, before, we were syncing in the wrong direction.
Now, the deposit bucket `self`, the source, will be synced with the published bucket `snapshot`, the destination.

Reply:
Exactly